src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs

```diff
@@ -169,6 +169,26 @@ public static IEnumerable<string> EnumerateRequiredProperties(string modelName,
             return propertiesList;
         }
 
+        /// <summary>
+        /// Returns the cumulative list of all discrimintor properties found in the entire model hierarchy
+        /// </summary>discrimintor
+        /// <param name="modelName">model for which to check the discrimintor properties</param>
+        /// <param name="definitions">dictionary of model definitions</param>
+        public static IEnumerable<string> EnumerateDiscrimintorProperties(string modelName, Dictionary<string, Schema> definitions)
+        {
+            var modelsToCheck = EnumerateModelHierarchy(modelName, definitions);
+            var propertiesList = new List<string>();
+            foreach (var modelRef in modelsToCheck)
+            {
+                if (!definitions.ContainsKey(modelRef) || definitions[modelRef].Discriminator?.Any() != true) {
```

Comment:
Why use Any()? use String.IsNullOrEmpty(String)  is more clear for string type.

Reply:
Good suggestion!